### PR TITLE
doc: releases: v3.5.0: zbus release notes

### PR DESCRIPTION
**Pull request by rodrigopex**
Add the changes of zbus during the Zephyr v3.5.0 development.

### CODE REVIEW

**Review by kartben — DISMISSED** [review]

**Comment by jhedberg** [comment]
@rodrigopex there seems to be a doc build CI failure, could you take a look?

File: doc/releases/release-notes-3.5.rst [hunk 1]
```diff
@@ -896,6 +896,29 @@ Libraries / Subsystems
...
+    the heap.
+  * Added a new type of observer called Message Subscriber. ZBus' VDED will send a copy of the
+    message during the publication/notification process.
+  * Changed the VDED delivery sequence. Check the ref:`documentation<zbus delivery sequence>`.
+  * ZBus runtime observers now rely on the heap instead of a memory pool.
+  * Added new iterable section iterators APIs (for channels and observers) can now receive a
+    ``user_data`` pointer to keep context between the function calls.
+  * Added APIs :c:macro:`ZBUS_LISTENER_DEFINE_WITH_ENABLE` and
+    :c:macro:`ZBUS_SUBSCRIBER_DEFINE_WITH_ENABLE` that enable developers to define disabled
+    observers. They need to be enabled during runtime to receive notifications from the bus.
```
Comment (kartben) [inline doc/releases/release-notes-3.5.rst:919]: fwiw this, to me, kind of confirms that the naming is not ideal.
"Use XXX_WITH_**ENABLE** to define a **disabled** observer" is really odd.

**Review by kartben — DISMISSED** [review]

File: doc/releases/release-notes-3.5.rst (hunk 1 above)
Comment (rodrigopex) [reply doc/releases/release-notes-3.5.rst:919]: Do you think there is still time to change that?

Comment (rodrigopex) [reply doc/releases/release-notes-3.5.rst:919]: The description could be better. The API intends to define the "enableness" of an observer programmatically.

**Comment by jhedberg** [comment]
@rodrigopex needs to be rebased (sorry for the hassle, this happens since many release notes PRs are getting merged right now)

**Comment by rodrigopex** [comment]
> @rodrigopex needs to be rebased (sorry for the hassle, this happens since many release notes PRs are getting merged right now)

Done. No problem.

**Review by jhedberg — APPROVED** [review]

**Review by kartben — APPROVED** [review]

File: doc/releases/release-notes-3.5.rst (hunk 1 above)
Comment (kartben) [reply doc/releases/release-notes-3.5.rst]: Yes, better now, thank you!

**Pull request merged**